### PR TITLE
Parse default config from env

### PR DESCRIPTION
**Pull request by ifuller1**
* also add watch support for test support

### CODE REVIEW

File: environment.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    let configElement = {};
+    configElement[key] = environmentVariable === undefined ? defaultConfigValue
+      : environmentVariable;
+
+    return {...previousValue, ...configElement};
+  }, {});
+}
+
+function getEnvironmentValueForObjectKey(key:string, originalValue:Object) {
+  let envKeyName = key.replace(/([A-Z])/g,
```
Comment (lucygilbert) [inline environment.js]: I'd suggest changing this regex to `/([A-Z]+)/` because config keys like `skipCollaboratorPR` become `SKIP_COLLABORATOR_PR`, rather than `SKIP_COLLABORATOR_P_R` which seems a bit strange.

File: __tests__/environment-test.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    })
+
+    expect(parsedConfig).toEqual({
+      aKeyName:'fromEnv'
+    });
+  })
+
+  describe('when parsing the env variable', () => {
+    afterEach(() => {
+        process.env = {};
```
Comment (lucygilbert) [inline __tests__/environment-test.js]: Should be indented 6 spaces not 8

File: environment.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      return rawValue;
+    case "number":
+      return Number.parseInt(rawValue);
+    case "boolean":
+      return rawValue.toString().toLowerCase() === 'true';
+    case "object":
+      if( Array.isArray(originalValue) ) {
+        return rawValue.split(",").map((item)=>item.trim())
+      } else {
+        return rawValue;
```
Comment (lucygilbert) [inline environment.js]: Might this raw value of the environment variable be a string representation of the object sometimes? Maybe `try` to `JSON.parse` it, and fall back to just the `rawValue` in the `catch`?

File: package.json [hunk 4]
```diff
@@ -4,13 +4,22 @@
   "license": "BSD-3-Clause",
   "repository": "facebook/mention-bot",
   "scripts": {
-    "test": "flow && jest",
-    "start": "node run-server.js"
+     "test": "flow && jest",
+     "start": "node run-server.js",
+     "watch": "npm-watch"
```
Comment (lucygilbert) [inline package.json]: So, this watch is supposed to run in the background and run the tests every time the JS is changed? If so, this change is unrelated to the subject of your PR and the subject of the commit message.

File: environment.js (hunk 3 above)
Comment (ifuller1) [inline environment.js]: I did think about this but the only default config types, at the moment, so I went for YAGNI. 

Just about to push the other changes...

File: environment.js (hunk 1 above)
Comment (ifuller1) [inline environment.js]: adding a test to cover this
